core/src/main/java/hudson/PluginManager.java

```diff
@@ -137,7 +137,25 @@
     /**
      * All active plugins, topologically sorted so that when X depends on Y, Y appears in the list before X does.
      */
-    protected final List<PluginWrapper> activePlugins = new CopyOnWriteArrayList<PluginWrapper>();
+    protected final List<PluginWrapper> activePlugins = new CopyOnWriteArrayList<PluginWrapper>() {
+        @Override
+        public boolean add(PluginWrapper pluginWrapper) {
+            try {
+                return super.add(pluginWrapper);
+            } finally {
+                onActivate(pluginWrapper);
```

Comment:
_Just a nit_ Since the event is fired after the plugin has been added it would suggest using a past tense verb in the interface instead.